Assets/HoloToolkit/Input/Scripts/Cursor/Cursor.cs

```diff
@@ -306,11 +306,29 @@ protected virtual void OnPointerSpecificFocusChanged(IPointingSource pointer, Ga
         /// Override function when a new modifier is found or no modifier is valid
         /// </summary>
         /// <param name="modifier"></param>
+        [System.Obsolete("Use OnActiveModifier(ICursorModifier modifier) instead.")]
         protected virtual void OnActiveModifier(CursorModifier modifier)
         {
             TargetedCursorModifier = modifier;
```

Comment:
Shouldn't this now be calling the new method with the interface parameter?

Reply:
Following is the OnPointerSpecificFocusChanged method:
![onpointerspecificfocuschanged](https://user-images.githubusercontent.com/14876006/43379037-039d5814-93e8-11e8-8661-d262a0f3b997.PNG)

Here, if I call the old OnActiveModifier(CursorModifier modifier) method instead of the new one with interface parameter, I need to cast the ICursorModifier object to CursorModifer which will return null in all the cases when newFocusedObject has an implementation of ICursorModifier which is not CursorModifier.
Hence, the old OnActiveModifier(CursorModifier modifier) method will receive null as a parameter and pass null to the new implementation of the method.

In order to avoid this issue, I am calling the new OnActiveModifier with the interface parameter from OnPointerSpecificFocusChanged since ICursorModifier is the parent here and from inside of this method, I am checking if this implementation of ICursorModifier is CursorModifier, only then I call the old OnActiveModifier.
![new_onactivemodifier](https://user-images.githubusercontent.com/14876006/43379311-61dc2594-93e9-11e8-9b4c-e19ef031c6b1.PNG)


This keeps the old functionality intact and will not give any errors or produce any unexpected behavior for those users who have custom classes inherited from these.

Reply:
@StephenHodgson Are we good with this?